sycl/include/sycl/vector_preview.hpp

```diff
@@ -983,13 +983,21 @@ template <typename Type, int NumElements> class vec {
     }
   }
 
-#ifdef __SYCL_BINOP
-#error "Undefine __SYCL_BINOP macro"
+#if defined(__SYCL_BINOP) || defined(BINOP_BASE)
+#error "Undefine __SYCL_BINOP and BINOP_BASE macro"
 #endif
 
-#ifdef __SYCL_USE_EXT_VECTOR_TYPE__
-#define __SYCL_BINOP(BINOP, OPASSIGN, CONVERT)                                 \
-  friend vec operator BINOP(const vec &Lhs, const vec &Rhs) {                  \
+// Note: vec<>/SwizzleOp logical value is 0/-1 logic, as opposed to 0/1 logic.
+// As far as CTS validation is concerned, 0/-1 logic also applies when
+// NumElements is equal to one, which is somewhat inconsistent with being
+// transparent with scalar data.
+// TODO: Determine if vec<, NumElements=1> is needed at all, remove this
+// inconsistency if not by disallowing one-element vectors (as in OpenCL)
+#ifdef __SYCL_DEVICE_ONLY__
+#define BINOP_BASE(BINOP, OPASSIGN, CONVERT, COND)                             \
+  template <typename T = DataT>                                                \
+  friend typename std::enable_if_t<(COND), vec> operator BINOP(                \
+      const vec & Lhs, const vec & Rhs) {                                      \
```

Comment:
```suggestion
      const vec &Lhs, const vec &Rhs) {                                        \
```

Reply:
`clang-format` wants `const vec & Lhs` instead of `const vec &Lhs`